aspnetcore/fundamentals/hosting.md

```diff
@@ -1,225 +1,530 @@
+    ...
+```
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+```csharp
+var host = new WebHostBuilder()
+    .UseSetting("shutdownTimeoutSeconds", "10")
+    ...
 ```
 
-In ASP.NET Core 2.0, Kestrel has its own endpoint configuration API and does not support `https://` in the `urls` string. For more information, see [Introduction to Kestrel](xref:fundamentals/servers/kestrel?tabs=aspnetcore2x#endpoint-configuration).
+---
+
+### Startup Assembly
 
-**Startup Assembly** `string`
+Determines the assembly to search for the `Startup` class.
 
-Key: `startupAssembly`. Determines the assembly to search for the `Startup` class. Set using the `UseStartup` method. May instead reference specific type using `WebHostBuilder.UseStartup<StartupType>`. If multiple `UseStartup` methods are called, the last one takes precedence.
+**Key**: startupAssembly  
+**Type**: *string*  
+**Default**: The app's assembly  
+**Set using**: `UseStartup`
+
+You can reference the assembly by name (`string`) or type (`TStartup`). If multiple `UseStartup` methods are called, the last one takes precedence.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+```csharp
+WebHost.CreateDefaultBuilder(args)
+    .UseStartup("StartupAssemblyName")
+    ...
+```
+
+```csharp
+WebHost.CreateDefaultBuilder(args)
+    .UseStartup(<TStartup>)
+    ...
+```
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 ```csharp
-new WebHostBuilder()
+var host = new WebHostBuilder()
     .UseStartup("StartupAssemblyName")
+    ...
+```
+
+```csharp
+var host = new WebHostBuilder()
+    .UseStartup(<TStartup>)
+    ...
 ```
 
-<a name=web-root-setting></a>
+---
+
+### Web Root
+
+Sets the relative path to the app's static assets.
+
+**Key**: webroot  
+**Type**: *string*  
+**Default**: If not specified, the default is "(Content Root)/wwwroot", if the path exists. If the path doesn't exist, then a no-op file provider is used.  
+**Set using**: `UseWebRoot`
 
-**Web Root** `string`
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+```csharp
+WebHost.CreateDefaultBuilder(args)
+    .UseWebRoot("public")
+    ...
+```
 
-Key: `webroot`. If not specified the default is `(Content Root Path)\wwwroot`, if it exists. If this path doesn't exist, then a no-op file provider is used. Set using `UseWebRoot`.
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 ```csharp
-new WebHostBuilder()
+var host = new WebHostBuilder()
     .UseWebRoot("public")
+    ...
 ```
 
-### Overriding Configuration
+---
+
+## Overriding configuration
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+When you use `CreateDefaultBuilder` to build a host, you automatically enable configuration for *appsettings.json*, *appsettings.{Environment}.json*, [user secrets](xref:security/app-secrets) (in `Development`), environment variables, and command-line arguments. If you wish to add additional configuration sources, such as optional configuration stored in a *hosting.json* file, you can build additional configuration with `UseConfiguration`.
```

Comment:
This subject is a bit confusing as there are two stages to config. There is host config that starts with ASPNETCORE_* environment variables, WebHostBuilder.UseConfiguration and UseSettings adds to this. The host and server settings described above are resolved from this dataset.

App configuration is initialized with the final host configuration and then ConfigureAppConfiguration builds on top of that. There's a known issue where the CreateDefaultBuilder args parameter is only added to the app level config, not the host config. https://github.com/aspnet/MetaPackages/issues/221

Reply:
Your samples aren't wrong, but we should be explicit which config is being updated.

Reply:
lol ... I realize that the start of this paragraph reflects *what I wish to be true* and not *what's actually the case*. I *wish* it were all just *one config* (KISS) ... a world where the base and app configs are *one configuration* and last setting for anything wins. I understand it isn't that simple (e.g., reloading).

The beginning of this paragraph is blown because, as you say, the app config is implied as possibly being a source of the base config. They are conflated. I'll modify this to concentrate on the concept conveyed by last sentence and get rid of the rest.